138828505 add jira integration resource v5

tmikeschu opened the pull request:
Add resource for jira_integration endpoint. All tests passing and rebased to current v5.

tmikeschu committed:
Add JiraIntegration resource and unit test

Merge conflicts

[#138828505]

tmikeschu committed:
Add attributes to story

Owner_ids and owned_by_id

[#138828505]

jasonnoble commented on lib/pivotal-tracker/jira_integration.rb:
```diff
@@ -0,0 +1,15 @@
+class PivotalTracker::JiraIntegration
```
Please inherit from our new PivotalTracker::Resource class (introduced as part of https://github.com/jasonnoble/pivotal-tracker/pull/11/files#diff-db66140f03dd10afa01447b572c04b2dR1)

tmikeschu replied:
👍

jasonnoble commented on lib/pivotal-tracker/story.rb:
```diff
@@ -1,12 +1,12 @@
 class PivotalTracker::Story
```
Since you're in here...  Please inherit from our new PivotalTracker::Resource class (introduced as part of https://github.com/jasonnoble/pivotal-tracker/pull/11/files#diff-db66140f03dd10afa01447b572c04b2dR1)

tmikeschu replied:
👍

jasonnoble commented on spec/fixtures/pivotal_tracker_cassettes/account.yml:
```diff
@@ -2,7 +2,7 @@
 http_interactions:
 - request:
     method: get
-    uri: https://www.pivotaltracker.com/services/v5/accounts//memberships
```
I don't think we need these changes.

jasonnoble commented on spec/fixtures/pivotal_tracker_cassettes/fetch_token.yml:
```diff
@@ -116,24 +41,23 @@ http_interactions:
-      string: '{"kind":"me","id":861657,"name":"Pivotal Tracker API Gem","initials":"PTAG","username":"<PIVOTAL_USER_NAME>","time_zone":{"kind":"time_zone","olson_name":"America/Los_Angeles","offset":"-08:00"},"api_token":"<PIVOTAL_API_TOKEN>","has_google_identity":false,"projects":[{"kind":"membership_summary","id":2620631,"project_id":708251,"project_name":"Pivotal
-        Tracker API Gem","project_color":"74a4d7","favorite":false,"role":"owner","last_viewed_at":"2017-01-30T18:28:44Z"},{"kind":"membership_summary","id":2620635,"project_id":708253,"project_name":"My
-        new tracker project","project_color":"e46642","favorite":false,"role":"owner","last_viewed_at":"2012-12-10T06:14:35Z"}],"email":"<PIVOTAL_EMAIL_ADDRESS>","receives_in_app_notifications":true,"created_at":"2012-12-10T02:08:07Z","updated_at":"2017-01-30T20:35:45Z"}'
+      string: '{"kind":"me","id":2415615,"name":"Mike Schutte","initials":"TMS","username":"<PIVOTAL_USER_NAME>","time_zone":{"kind":"time_zone","olson_name":"America/Los_Angeles","offset":"-08:00"},"api_token":"<PIVOTAL_API_TOKEN>","has_google_identity":false,"projects":[{"kind":"membership_summary","id":8190509,"project_id":<PROJECT_ID>,"project_name":"Gem
```
Probably don't need these either.

jasonnoble commented on spec/fixtures/pivotal_tracker_cassettes/profile.yml:
```diff
@@ -120,24 +41,23 @@ http_interactions:
-      string: '{"kind":"me","id":861657,"name":"Pivotal Tracker API Gem","initials":"PTAG","username":"<PIVOTAL_USER_NAME>","time_zone":{"kind":"time_zone","olson_name":"America/Los_Angeles","offset":"-08:00"},"api_token":"<PIVOTAL_API_TOKEN>","has_google_identity":false,"projects":[{"kind":"membership_summary","id":2620631,"project_id":708251,"project_name":"Pivotal
-        Tracker API Gem","project_color":"74a4d7","favorite":false,"role":"owner","last_viewed_at":"2017-01-30T18:28:44Z"},{"kind":"membership_summary","id":2620635,"project_id":708253,"project_name":"My
-        new tracker project","project_color":"e46642","favorite":false,"role":"owner","last_viewed_at":"2012-12-10T06:14:35Z"}],"email":"<PIVOTAL_EMAIL_ADDRESS>","receives_in_app_notifications":true,"created_at":"2012-12-10T02:08:07Z","updated_at":"2017-01-30T20:35:45Z"}'
+      string: '{"kind":"me","id":2415615,"name":"Mike Schutte","initials":"TMS","username":"<PIVOTAL_USER_NAME>","time_zone":{"kind":"time_zone","olson_name":"America/Los_Angeles","offset":"-08:00"},"api_token":"<PIVOTAL_API_TOKEN>","has_google_identity":false,"projects":[{"kind":"membership_summary","id":8190509,"project_id":<PROJECT_ID>,"project_name":"Gem
```
Don't need these either...

jasonnoble commented on spec/fixtures/pivotal_tracker_cassettes/project.yml:
```diff
@@ -286,32 +229,167 @@ http_interactions:
+        release resource","description":"Provide attr_accessors and initialize method
+        based on existing models.","current_state":"unstarted","requested_by_id":2415615,"url":"https://www.pivotaltracker.com/story/show/138816383","project_id":<PROJECT_ID>,"owner_ids":[2413445],"labels":[],"owned_by_id":2413445},{"kind":"story","id":138828543,"created_at":"2017-02-01T15:44:07Z","updated_at":"2017-02-01T18:22:51Z","estimate":1,"story_type":"feature","name":"Add
+        lighthouse_integration","description":"Provide attr_accessors and initialize
+        method based on existing models.","current_state":"unstarted","requested_by_id":2413445,"url":"https://www.pivotaltracker.com/story/show/138828543","project_id":<PROJECT_ID>,"owner_ids":[2413445],"labels":[],"owned_by_id":2413445},{"kind":"story","id":138755539,"created_at":"2017-01-31T21:02:55Z","updated_at":"2017-01-31T21:02:55Z","story_type":"feature","name":"Add
```
Were these changes intentional?

tmikeschu replied:
@jasonnoble I was having some authentication issues. How should I go about hiding these?

jasonnoble replied:
Amend your commit, and don't stage the cassette changes...

tmikeschu replied:
✔️

tmikeschu committed:
Refactor model initializers

Inherit from Resource model for initialize method.

[#138828505]

tmikeschu committed:
Add attributes for AccountMembership

Note that :person is brought in from the API call, but not on the doc
list.

[#138828505]
